File: frontend/src/shared/api/rewardApi.ts
import { apiClient } from "@/shared/api/client";

export interface WeeklyScoreResponse {
  id: string;
  week_start: string;
  record_days_score: number | null;
  receipt_score: number | null;
  diary_score: number | null;
  budget_score: number | null;
  streak_score: number | null;
  total_score: number | null;
  reward_granted: boolean | null;
}

export interface StreakResponse {
  current_streak: number | null;
  longest_streak: number | null;
  last_record_date: string | null;
}

export interface BoxCountResponse {
  box_count: number;
  daily_earned: number; // 오늘 당일 영수증 인증으로 받은 뽑기권 수
  daily_limit: number; // 하루 최대 (당일 영수증 인증) 뽑기권 수
}

export interface RewardBoxItemResponse {
  inventory_id: string;
  item_id: string;
  name: string;
  image_url: string;
  is_equipped: boolean;
  slot_name: string;
}

export type RewardType = "miss" | "spt" | "avatar";

export interface OpenBoxResponse {
  remaining_box_count: number;
  reward_type: RewardType;
  is_win: boolean;
  message: string;
  spt_amount: number | null;
  item: RewardBoxItemResponse | null;
}

export async function getCurrentWeeklyScore(): Promise<WeeklyScoreResponse> {
  const res = await apiClient.get<WeeklyScoreResponse>(
    "/api/rewards/weekly-score/current"
  );
  return res.data;
}

export async function getStreak(): Promise<StreakResponse> {
  const res = await apiClient.get<StreakResponse>("/api/rewards/streak");
  return res.data;
}

export async function getRewardBoxCount(): Promise<BoxCountResponse> {
  const res = await apiClient.get<BoxCountResponse>("/api/unity/rewards/box-count");
  return res.data;
}

export async function openRewardBox(): Promise<OpenBoxResponse> {
  const res = await apiClient.post<OpenBoxResponse>("/api/unity/rewards/open-box");
  return res.data;
}

File: android/app/src/main/java/com/ict/spentopia/data/remote/RewardApi.kt
package com.ict.spentopia.data.remote // 이 파일이 속한 패키지 위치를 적음

import retrofit2.http.GET // GET API 표시를 가져옴

data class WeeklyScoreResponse( // WeeklyScoreResponse 데이터를 묶어둘 클래스 시작
    val id: String?, // 아이디를 저장함
    val week_start: String?, // week_start 값을 저장함
    val record_days_score: Int?, // record_days_score 값을 저장함
    val receipt_score: Int?, // receipt_score 값을 저장함
    val diary_score: Int?, // diary_score 값을 저장함
    val budget_score: Int?, // 예산 관련 값을 저장함
    val streak_score: Int?, // streak_score 값을 저장함
    val total_score: Int?, // total_score 값을 저장함
    val reward_granted: Boolean? // reward_granted 값을 저장함
)

interface RewardApi { // RewardApi에서 꼭 만들어야 할 함수 규칙을 정함
    @GET("/api/rewards/weekly-score/current") // 서버에서 데이터를 가져오는 API 주소를 적음
    suspend fun getCurrentWeeklyScore(): WeeklyScoreResponse // 데이터를 불러오는 함수 시작
}
